FWCore/Utilities/src/ObjectWithDict.cc

```diff
@@ -74,9 +76,13 @@ namespace edm {
     }
 
     if (to.hasBase(from)) { // down cast
+#ifndef _LIBCPP_VERSION
       // use the internal dynamic casting of the compiler (e.g. libstdc++.so)
       void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
```

Comment:
/Users/gartung/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fwlite-9.0.X-ywgcs6iqugqoum574jaso7syo63lyzg2/CMSSW_9_0_X/src/FWCore/Utilities/src/ObjectWithDict.cc:81:70: error: no type named '__class_type_info' in namespace '__cxxabiv1'
      void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
                                                                ~~~~~^
/Users/gartung/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fwlite-9.0.X-ywgcs6iqugqoum574jaso7syo63lyzg2/CMSSW_9_0_X/src/FWCore/Utilities/src/ObjectWithDict.cc:81:87: error: expected '>'
      void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
                                                                                      ^
                                                                                      >
/Users/gartung/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fwlite-9.0.X-ywgcs6iqugqoum574jaso7syo63lyzg2/CMSSW_9_0_X/src/FWCore/Utilities/src/ObjectWithDict.cc:81:64: note: to match this '<'
      void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
                                                               ^
/Users/gartung/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fwlite-9.0.X-ywgcs6iqugqoum574jaso7syo63lyzg2/CMSSW_9_0_X/src/FWCore/Utilities/src/ObjectWithDict.cc:81:132: error: no type named '__class_type_info' in namespace '__cxxabiv1'
      void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
                                                                                                                              ~~~~~^
/Users/gartung/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fwlite-9.0.X-ywgcs6iqugqoum574jaso7syo63lyzg2/CMSSW_9_0_X/src/FWCore/Utilities/src/ObjectWithDict.cc:81:149: error: expected '>'
      void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
                                                                                                                                                    ^
                                                                                                                                                    >
/Users/gartung/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fwlite-9.0.X-ywgcs6iqugqoum574jaso7syo63lyzg2/CMSSW_9_0_X/src/FWCore/Utilities/src/ObjectWithDict.cc:81:126: note: to match this '<'
      void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
                                                                                                                             ^
/Users/gartung/spack/opt/spack/darwin-sierra-x86_64/clang-8.0.0-apple/fwlite-9.0.X-ywgcs6iqugqoum574jaso7syo63lyzg2/CMSSW_9_0_X/src/FWCore/Utilities/src/ObjectWithDict.cc:81:28: error: no member named '__dynamic_cast' in namespace '__cxxabiv1'
      void* address = abi::__dynamic_cast(address_, static_cast<abi::__class_type_info const*>(&from.typeInfo()), static_cast<abi::__class_type_info const*>(&to.typeInfo()), -1);
                      ~~~~~^
5 errors generated.